### PR TITLE
Fix the Chocolatey-missing error again

### PR DESCRIPTION
**Pull request by randomcamel**
The previous implementation (in `define_resource_requirements`) failed acceptance testing because `choco.exe` gets called before that, in `load_current_resource`, so the detection needs to be a bit lower-level.

This includes a spec that replicates the acceptance test (by calling `load_current_resource`).

### CODE REVIEW

File: lib/chef/provider/package/chocolatey.rb [hunk 1]
```diff
@@ -152,11 +147,15 @@ def check_resource_semantics!
...
-          @choco_exe ||= ::File.join(
-            choco_install_path,
-            "bin",
-            "choco.exe"
-            )
+          @choco_exe ||=
+              # if this check is in #define_resource_requirements, it won't get
+              # run before choco.exe gets called from #load_current_resource.
+              exe_path = ::File.join(choco_install_path.to_s, "bin", "choco.exe")
+          if !::File.exist?(exe_path)
```
Comment (mwrock) [inline lib/chef/provider/package/chocolatey.rb]: looks like some indentation issues here. Also `unless` may be clearer than `if !`.

Comment (lamont-granquist) [inline lib/chef/provider/package/chocolatey.rb]: yeah that'll be a rubocop violation at some point in the not too distant future...

File: lib/chef/provider/package/chocolatey.rb [hunk 2]
```diff
@@ -152,11 +147,15 @@ def check_resource_semantics!
...
+            raise Chef::Exceptions::MissingLibrary, CHOCO_MISSING_MSG
+          else
+            exe_path
+          end
```
Comment (lamont-granquist) [inline lib/chef/provider/package/chocolatey.rb]: could we do this a bit more like...

```
  @choco_exe ||=
    begin
      exe_path = ...
      raise unless exist?
      exe_path
   end
```

its both cleaner to read, and also if we've set choco_exe successfully we don't need to re-test if we should raise again the next time we hit it, and the return value buried inside the if is just a bit hard to read (much like this run on sentence...)

File: lib/chef/provider/package/chocolatey.rb [hunk 3]
```diff
@@ -152,11 +147,15 @@ def check_resource_semantics!
...
         def choco_exe
-          @choco_exe ||= ::File.join(
-            choco_install_path,
-            "bin",
-            "choco.exe"
-            )
+          @choco_exe ||=
+              # if this check is in #define_resource_requirements, it won't get
+              # run before choco.exe gets called from #load_current_resource.
+              exe_path = ::File.join(choco_install_path.to_s, "bin", "choco.exe")
```
Comment (randomcamel) [inline lib/chef/provider/package/chocolatey.rb]: Somehow `exe_path` comes up `nil` here sometimes.

File: lib/chef/provider/package/chocolatey.rb (hunk 2 above)
Comment (randomcamel) [inline lib/chef/provider/package/chocolatey.rb]: Yep, good call.

**Comment by mwrock** [comment]
:+1:

**Comment by btm** [comment]
:+1:

**Pull request merged**